pkg/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -1296,8 +1320,35 @@ func (n *NetworkPolicyController) syncAddressGroup(key string) error {
 		internalNP := internalNPObj.(*antreatypes.NetworkPolicy)
 		addrGroupNodeNames = addrGroupNodeNames.Union(internalNP.SpanMeta.NodeNames)
 	}
+	addressGroupObj, found, _ := n.addressGroupStore.Get(key)
```

Comment:
Not sure why it's moved from L1284 to here. addrGroupNodeNames will not be used any if the group is deleted so the previous code seems more efficient.